ydb/library/actors/core/event_pb.cpp

```diff
@@ -221,4 +224,213 @@ namespace NActors {
+                return Max<size_t>();
+            }
+            const char byte = *iter.ContiguousData();
+            iter += 1;
+            --size;
+            res |= (static_cast<size_t>(byte) & 0x7F) << offset;
+            offset += 7;
+            if (!(byte & 0x80)) {
+                break;
+            }
+        }
+        return res;
+    }
+
+    bool SerializeToArcadiaStreamImpl(TChunkSerializer* chunker, const TVector<TRope> &payload) {
+        // serialize payload first
+        if (payload) {
+            void *data;
+            int size = 0;
+            auto append = [&](const char *p, size_t len) {
+                while (len) {
+                    if (size) {
+                        const size_t numBytesToCopy = std::min<size_t>(size, len);
+                        memcpy(data, p, numBytesToCopy);
+                        data = static_cast<char*>(data) + numBytesToCopy;
+                        size -= numBytesToCopy;
+                        p += numBytesToCopy;
+                        len -= numBytesToCopy;
+                    } else if (!chunker->Next(&data, &size)) {
+                        return false;
+                    }
+                }
+                return true;
+            };
+            auto appendNumber = [&](size_t number) {
+                char buf[MaxNumberBytes];
+                return append(buf, SerializeNumber(number, buf));
+            };
+
+#if USE_EXTENDED_PAYLOAD_FORMAT
+            char marker = ExtendedPayloadMarker;
+            append(&marker, 1);
+            if (!appendNumber(payload.size())) {
+                return false;
+            }
+            for (const TRope& rope : payload) {
+                if (!appendNumber(rope.GetSize())) {
+                    return false;
+                }
+            }
+            if (size) {
+                chunker->BackUp(std::exchange(size, 0));
+            }
+            for (const TRope& rope : payload) {
+                if (!chunker->WriteRope(&rope)) {
+                    return false;
+                }
+            }
+#else
+            char marker = PayloadMarker;
+            append(&marker, 1);
+            if (!appendNumber(payload.size())) {
+                return false;
+            }
+            for (const TRope& rope : payload) {
+                if (!appendNumber(rope.GetSize())) {
+                    return false;
+                }
+                if (rope) {
+                    if (size) {
+                        chunker->BackUp(std::exchange(size, 0));
+                    }
+                    if (!chunker->WriteRope(&rope)) {
+                        return false;
+                    }
+                }
+            }
+            if (size) {
+                chunker->BackUp(size);
+            }
+#endif
+        }
+
+        return true;
+    }
+
+
+    void FillExtendedFormat(TRope::TConstIterator &iter, size_t &size, TVector<TRope> &payload, size_t &totalPayloadSize)
```

Comment:
"FillExtendedFormat" does not really describe what it does -- it's better to name it "ParseExtendedFormatPayload"